service-report/src/main/kotlin/org/veupathdb/service/mblast/report/db/sql/select-report-jobs.kt

```diff
@@ -0,0 +1,88 @@
+package org.veupathdb.service.mblast.report.db.sql
+
+import org.intellij.lang.annotations.Language
+import org.veupathdb.lib.hash_id.HashID
+import org.veupathdb.service.mblast.report.db.*
+import java.sql.Connection
+import java.sql.ResultSet
+
+
+// ////////////////////////////////////////////////////////////////////////// //
+//                                                                            //
+//  Select by User ID                                                         //
+//                                                                            //
+// ////////////////////////////////////////////////////////////////////////// //
+
+@Language("Oracle")
+private const val SQL_BY_USER = """
+SELECT
+  a.${Column.ReportJobID}
+, a.${Column.QueryJobID}
+, a.${Column.Config}
+, a.${Column.CreatedOn}
+, b.${Column.UserID}
+, b.${Column.Summary}
+, b.${Column.Description}
+FROM
+  ${Schema.MBlast}.${Table.ReportConfigs} a
+  INNER JOIN ${Schema.MBlast}.${Table.ReportToUsers} b
+    ON a.${Column.ReportJobID} = b.${Column.ReportJobID}
+WHERE
+  b.${Column.UserID} = ?
+"""
+
+/**
+ * Fetches all report jobs associated with the given [userID].
+ *
+ * @param userID ID of the user whose report jobs should be fetched.
+ *
+ * @return A list of zero or more report jobs associated with the target user.
+ */
+fun Connection.selectReportJobs(userID: Long) =
+  prepareStatement(SQL_BY_USER).use { ps ->
+    ps.setLong(1, userID)
+    ps.executeQuery().use { it.toList(fn = ResultSet::parseUserReportDetails) }
+  }
```

Comment:
If this is an unbounded list, do we need to add pagination here?

Reply:
Not sure if pagination is overkill or not for several reasons.

* I suspect that it is very unlikely for the list to grow in size even to 100, it is technically possible for someone to go HAM on our API and get a list that is too big for the client to handle, but even then to get a list too big for the _service_ to handle would still be a feat.

* What is the point at which we are sending "too much" data to the client?

* Is pagination on list endpoints just a good standard practice that all our list endpoints should follow?

Reply:
My inclination is to always add it because it can be hard to add later after you release to the world. It actually doesn't really matter here since we are likely the only client so I'm happy to defer for now!